tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/work_order_singleton.cpp

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "enclave_u.h"
+
+#include "avalon_sgx_error.h"
+#include "log.h"
+#include "types.h"
+
+#include "enclave.h"
+#include "base.h"
+#include "work_order_singleton.h"
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+tcf_err_t WorkOrderHandlerSingleton::HandleWorkOrderRequest(
```

Comment:
Is workroder handler part is same for singleton and WPE right? Then we common out the work order handler part?

Reply:
No, they are different for different workers. Only common API is GetSerializedResponse, hence its made part of base class.